httpserver/bin/mini_file_server.dart

```diff
@@ -9,20 +9,35 @@
 
 import 'dart:io';
 
-main() {
-  HttpServer.bind(InternetAddress.LOOPBACK_IP_V4, 4044).then((server) {
-    server.listen((HttpRequest req) {
-      File file = new File('index.html');
-      file.exists().then((bool found) {
-        if (found) {
-          file.openRead()
-              .pipe(req.response)  // HttpResponse type.
-              .catchError((e) => print(e.toString()));
-        } else {
-          req.response.statusCode = HttpStatus.NOT_FOUND;
-          req.response.close();
-        }
-      });
-    });
-  });
+main() async {
+  var server;
+
+  try {
+    server =
+        await HttpServer.bind(InternetAddress.LOOPBACK_IP_V4, 4044);
+  } catch (e) {
+    print("Couldn't bind to port 4044: $e");
+    exit(-1);
+  }
+
+  await for (HttpRequest req in server) {
+    File file = new File('index.html');
+    if (await file.exists()) {
+      print("Serving index.html.");
+      req.response.headers.contentType = ContentType.HTML;
+      try {
+        await file
+            .openRead()
+            .pipe(req.response); // HttpResponse type.
```

Comment:
I'm not sure what the comment means or what it binds to.
If the type is important, make a variable for it, otherwise just lose the comment.